clippy_lints/src/methods/mod.rs

```diff
@@ -2752,6 +2780,10 @@ impl Methods {
                         unnecessary_lazy_eval::check(cx, expr, recv, arg, "or");
                     }
                 },
+                ("replace" | "replacen", [arg1, arg2] | [arg1, arg2, _]) => {
+                    no_effect_replace::check(cx, expr, arg1, arg2);
+                    collapsible_str_replace::check(cx, expr, name, recv);
+                },
```

Comment:
```suggestion
                (name @ "replace" | "replacen", [arg1, arg2] | [arg1, arg2, _]) => {
                    no_effect_replace::check(cx, expr, arg1, arg2);
                    if name == "replace" && let Some(("replace", ..)) = method_call(recv) {
                        collapsible_str_replace::check(cx, expr, name, recv);
                    }
                },
```

So something like this.